simulator/risc_v/t/unit_test.cpp

```diff
@@ -596,3 +605,21 @@ TEST_CASE("RISCV RV128 unshfl")
     }
 }
 
+TEST_CASE("RISCV ecall")
+{
+    RISCVInstr<uint32> instr( "ecall", 0);
+    instr.execute();
+    CHECK( instr.trap_type() == Trap::SYSCALL);
+}
+
+TEST_CASE("RISCV32 driver")
+{
+    auto sim = Simulator::create_configured_isa_simulator( "riscv32");
+    auto drv = create_riscv32_driver( sim.get());
+    CHECK( drv->handle_trap( get_op_with_trap( Trap( Trap::NO_TRAP))) == Trap::NO_TRAP);
+    CHECK( drv->handle_trap( get_op_with_trap( Trap( Trap::HALT))) == Trap::HALT);
+    auto expected_pc = (sim->read_csr_register( "stvec") >> 2U) & ~(bitmask<uint64>( 3));
+    CHECK( sim->get_pc() == expected_pc);
+    auto expected_cause = Trap( Trap::HALT).to_riscv_format();
+    CHECK( sim->read_csr_register( "scause") == expected_cause);
+}
```

Comment:
I would move that to a new file, since instruction test is very long already and takes a lot of time to compile.